storage/snapshot/snapshot.rs

```diff
@@ -456,8 +489,9 @@ impl<D: DurabilityClient> CommittableSnapshot<D> for WriteSnapshot<D> {
         }
     }
 
-    fn into_commit_record(self) -> CommitRecord {
-        CommitRecord::new(self.operations, self.open_sequence_number, CommitType::Data)
+    fn into_commit_record(mut self) -> (ReaderDropGuard, CommitRecord) {
```

Comment:
return the guard and the record so the caller can decide when to drop the reader handle (not the cleanest, but works)